Tweaks to comply with Cambridge BGS regs 2020

deryckchan opened the pull request:
Hardbound thesis submission formatting requirements from Cambridge University Board of Graduate Studies have changed since this template was last updated in 2018. Some of the changes could not be satisfied without modifying the class file, so this patch tweaked some details to provide functionality to comply with the latest Cambridge regulations.

Changes include:
- `abstract' option (abstract-only mode) now hides title page, nomenclature, and references as well (class file and thesis.tex)
- New class option `ntabstract' to print name and title onto abstract even when bound into the whole thesis (class file)
- Abstract now comes before Acknowledgements (thesis.tex)
- Default horizontal margins now defined using `width' rather than `ratio', so that bindingoffset no longer causes print and web editions to have different pagination. (class file)

I can update CHANGELOG and README as well if this pull request is committed into the master branch.

kks32 commented:
@deryckchan Thank you for the changes, much appreciated. I would like to verify the following, as the guidelines still include these in the submission: https://www.cambridgestudents.cam.ac.uk/your-course/examinations/graduate-exam-information/after-examination/degree-approval-and-1

It still says we need to submit a title page, declaration and the abstract. Could you please provide a link to the new guidelines where it says we don't need those?

Thanks again!

deryckchan commented:
Hello @kks32 ! It's the same guideline page that you linked to. Notice that "title page" and "declaration" are mentioned in the "Bound inside the thesis" section but not the "Loose with the thesis" section. Hence I've made `abstract` mode strip literally everything except the abstract.

Also point 4 of "Bound inside of the thesis" says "Summary/Abstract with your name and thesis title on it", and then the next section "Loose with the thesis" says "Summary/Abstract (the same as the one you have bound in to your thesis)". Hence the `ntabstract` toggle (so that people submitting for other degrees / institutions can still have an abstract section that says "Abstract" rather than their name if they wish). [Searching for `+latex +"ntabstract"`](https://duckduckgo.com/?q=%2Blatex+%2B%22ntabstract%22&t=h_&ia=web) returns precisely zero results so the option name shouldn't conflict with any package one may want to invoke alongside this thesis template.

deryckchan commented:
@kks32 Done, please review and merge.

kks32 commented on PhDThesisPSnPDF.cls:
```diff
@@ -224,6 +224,11 @@ footnotes.}
+% Declare new Boolean variable to allow separate toggle 
+% to include author name and title in abstract,
+% i.e. Cambridge BGS style
+\newif\ifntabstract\ntabstractfalse
```
```suggestion
\newif\ifPHD@abstracttitle\PHD@abstracttitletrue
\DeclareOption{abstracttitle}PHD@abstracttitletrue}
\DeclareOption{noabstractitle}{\PHD@abstracttitlefalse}
```

kks32 replied:
We should have abstract title on by default as per the university guidelines, so let's have it turned to true.

kks32 commented on PhDThesisPSnPDF.cls:
```diff
@@ -1114,14 +1124,18 @@ wish to left align your text}
     \renewcommand{\bibliography}[1]{\thispagestyle{empty}}
+    
+    % Turn on name and title on abstract (DYC 2020)
+    \ntabstracttrue
```
Since we have this on by default we don't need this.

kks32 commented on PhDThesisPSnPDF.cls:
```diff
@@ -1114,14 +1124,18 @@ wish to left align your text}
 \newenvironment{abstract} {
-\ifPHD@abstract
-% Separate abstract as per Student Registry guidelines
+\ifntabstract
```
Since this is set as default, we don't need it here

deryckchan replied:
This is still needed for the "compile the entire thesis and don't put my name and thesis title onto the abstract page" combination to work.

kks32 commented on thesis.tex:
```diff
@@ -31,6 +31,10 @@
 % `abstract': To generate only the title page and abstract page with
 % dissertation title and name, to submit to the Student Registry
 %
+% `ntabstract': To add name and title to abstract page 
```
`abstracttitle` and `noabstracttitle`

kks32 reviewed (changes requested):
@deryckchan Thank you so much, sorry to ask you for further changes. 

Since the university requirement is to have title in the abstract, let's make sure it's enabled by default and could you please use the same styling `\PHD@abstracttitle` similar to `\PHD@lineno` used previously in the template? 

Also, could you please add `abstracttitle` and `noabstracttitle` to the README as well. 

Thank you so much for your contribution!

deryckchan commented:
I'm slightly concerned I might mess up the mixture of comment / suggested changes. Do you mind doing all of them as suggested changes so I can click commit (to my patch)?

pwichmann commented:
Has this now been merged? If not, are there any conflicts / problems?

deryckchan commented:
> @deryckchan Thank you so much, sorry to ask you for further changes.
> 
> Since the university requirement is to have title in the abstract, let's make sure it's enabled by default and could you please use the same styling `\PHD@abstracttitle` similar to `\PHD@lineno` used previously in the template?

Happy new year! I've looked into this in more depth. I've changed the default to `\ntabstracttrue` and added a `nontabstract` option.

However, it was a conscious decision to use `\ifntabstract` rather than the `PHD@optionname` nomenclature in this case, because:
- Conditional variables with `@` in them cannot be accessed outside the files in which they were declared. I can imagine there will be users who want to include other conditional statements in their abstract page depending on which abstract page format it is;
- The variable name `abstracttitle` is used by a number of other LaTeX thesis templates for quite different meanings, e.g. [K-State thesis template](https://www.k-state.edu/grad/academics/etdr/template/docs/Instructions-for-Using-LaTeX-ETDR-Template.pdf); [tikz abstract box tutorial](https://blog.karssen.org/2009/09/14/using-tikz-to-generate-an-abstract-box-in-latex/), so I thought it would be best to avoid that name, and instead coin a new name that means "Name and Title on ABSTRACT".

deryckchan commented:
> Has this now been merged? If not, are there any conflicts / problems?

@pwichmann - Not yet, @kks32 and I are having a discussion (as you can see here) about default behaviour and variable naming. A working copy of the template including my amendments can be found in [my fork](https://github.com/deryckchan/phd-thesis-template/tree/deryckchan-cambgspatch2020).
